### ISSUE TITLE
setLocale() and getApiValue() error with sets

### ISSUE THREAD

**Issue by JTNMW** [bolt/core#1389]
After updating to rc19 (composer update) from rc 16, or creating a new composer project (rc19), adding content to a set field causes the following error on saving the record. Data is being persisted to the database.

![Screenshot 2020-05-20 at 16 21 55](https://user-images.githubusercontent.com/57093734/82549370-bbe31100-9b54-11ea-8b84-2e725b5b54c9.png)

second error then occurs when attempting to list content with set field content.
![Screenshot 2020-05-21 at 11 28 36](https://user-images.githubusercontent.com/57093734/82550596-b1c21200-9b56-11ea-9c73-d10b701ff023.png)

Details
-------

| Question                      | Answer
|----------------- | ------
| Relevant Bolt Version | bolt 4 rc 19
| Install type                   |  Composer install 
| BC Break                     | yes/no
| PHP version                 | 7.4 |
| Web server                  | Apache / Nginx / Built-in, version [ version ]
| For UX/UI issues        | Browser name and version

Reproduction
------------

If you're filing a bug, please describe how to reproduce it. Include as much
relevant information as possible, such as:

**Comment by I-Valchev** [comment]
@JTNMW I'm not able to reproduce this yet. Can you please share your content type definition? Is this happening when creating a new record, or editing an existing one, or both? thanks!

**Comment by JTNMW** [comment]
@I-Valchev This happens when both saving existing and creating new records if Set fields are present for the contenttype

These are the steps to replicate:

- new composer project install; add dummy fixtures

- add Set fields to pages in contenttype.yaml - (this is the only change made in the project)

- edit or create new record. Upon saving error occurs.

```
pages:
    name: Pages
    singular_name: Page
    fields:
        title:
            type: text
            class: large
            group: content
        slug:
            type: slug
            uses: title
        image:
            type: image
            label: Highlighted image
        teaser:
            type: html
            height: 150px
        marky:
            type: markdown
        body:
            type: html
            height: 300px
            allowtwig: true
        textarea:
            type: textarea
            postfix: "This is a plain text area. the contents will <em><strong>not</em></strong> be processed."
        checkbox:
            type: checkbox
            label: "This is a checkbox"
        template:
            type: templateselect
            filter: '*.twig'
        contentset:
            type: set
            label: Content Set
            group: Sets
            fields:
                image:
                    label: Image with description
                    type: image
                text:
                    label: Paragraph
                    type: text
    taxonomy: [ groups ]
    records_per_page: 5
    order: id
```
The error is occuring when using both SQlite and Mysql database configuration.
thanks

**Comment by JTNMW** [comment]
frontend error for Record after saving/creating new with Set Fields present.
![image](https://user-images.githubusercontent.com/57093734/82654410-961f4000-9c18-11ea-8a05-1fc38c78d11f.png)

**Comment by I-Valchev** [comment]
Resolved, see #1390
